src/modules/NewsList/index.tsx

```diff
@@ -224,29 +296,79 @@ export function NewsList(_props: NewsListProps): ReactElement {
                                     }
                                 }
                                 publishedDate
-                                source
                                 title
+                                ... on NewsContent {
+                                    newsSource {
+                                        id
+                                        name
+                                    }
+                                }
                             }
                         }
                     }
                 }
             }
         `,
+        mergeResults,
         requestPolicy: 'cache-and-network',
         variables: {
             filter: {
                 companyIds: state.company ? [state.company.id] : undefined,
                 contentTypes: [ContentType.News],
                 searchTerm: state.searchTerm,
             },
+            fromIndex: state.fromIndex,
+            size: listSize,
         },
     });
 
+    const onChangeSearch = useCallback<ChangeHandler<string>>(
+        (_event, change) => {
+            setState((s) => ({
+                ...s,
+                canRefetch: false,
+                fromIndex: 0,
+                lastRefetch: DateTime.now(),
+                searchTerm: change.value || '',
+            }));
+        },
+        [state]
+    );
+
+    const onLoadMore = useCallback<ChangeHandler<number>>(
+        (event, change) => handlers.fromIndex(event, change),
+        [state]
+    );
+
+    const onRefetch = useCallback(() => {
+        const isPaginating = state.fromIndex > 0;
+        setState((s) => ({
+            ...s,
+            canRefetch: false,
+            fromIndex: 0, // if the user paginated, resetting to 0 will fire a new query
+            lastRefetch: DateTime.now(),
+        }));
+        // Refetch if not paginating
+        if (!isPaginating) {
+            newsListQuery.refetch({ requestPolicy: 'cache-and-network' });
+        }
+    }, [state.fromIndex]);
+
     const onSelectCompany = useCallback<ChangeHandler<CompanyFilterResult>>(
-        (event, change) => {
+        (_event, change) => {
             const primaryQuote = getPrimaryQuote(change.value);
             bus?.emit('instrument-selected', { ticker: primaryQuote?.localTicker }, 'out');
-            handlers.company(event, change);
+            setState((s) => ({
+                ...s,
+                canRefetch: false,
+                company: change.value || undefined,
+                fromIndex: 0,
+                lastRefetch: DateTime.now(),
```

Comment:
Might be easier to leave all these handlers alone and just do a single `useEffect()` that updates the `canRefetch`, `fromIndex` and `lastRefetch` when any of the relevant things changes (searchTerm, company)